stl/inc/algorithm

```diff
@@ -4534,6 +4535,8 @@ namespace ranges {
             _STL_INTERNAL_STATIC_ASSERT(indirectly_copyable<_It, _Out>);
             _STL_INTERNAL_STATIC_ASSERT(indirect_binary_predicate<equal_to, projected<_It, _Pj>, const _Ty1*>);
 
+            _STD _Verify_ranges_do_not_overlap(_First, _Last, _Output);

```

Comment:
This check is being added to `_Replace_copy_unchecked()`, which doesn't feel right. I think that the checks should be added outside of any unchecked helpers.

Reply:
It is both centralized for all overloads and already unwrapped iterator.

Reply:
Though the check would unwrap to pointers on its own

Reply:
After looking into this, I'm asking for confirmation if `_unchecked` is really bad idea. We would need to extract iterators from the places like here and introduce variables, this would make the change more massive:
https://github.com/microsoft/STL/blob/da7307d68dbf39a805372e91a740393cbf06595c/stl/inc/algorithm#L4621-L4627

As an alternative, "unchecked' functions can be renamed to "common", like this one:
https://github.com/microsoft/STL/blob/da7307d68dbf39a805372e91a740393cbf06595c/stl/inc/algorithm#L5703-L5708

Reply:
I think I would be OK with renaming "unchecked" to "common". That would be a reminder to potential callers that some checking might be performed. We just want to avoid the situation where one function performs checks and then calls something that performs redundant checks.

(If the centralized function is private-static then that substantially mitigates my concerns too.)